debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/TypeRegistry.java

```diff
@@ -458,7 +458,12 @@ public int getSqlType(String typeName) throws SQLException {
             // get custom type mappings from the map which was built up with a single query
             else {
                 try {
-                    return sqlTypesByPgTypeNames.get(typeName);
+                    final Integer pgType = sqlTypesByPgTypeNames.get(typeName);
+                    if (pgType != null) {
```

Comment:
@jpechane, in which situation would this happen? Can you add a comment either here or in the original Jira issue?

Reply:
@gunnarmorling This will happen if a new postres type is created during the streaming so it is not available in the start cache.

Reply:
Ok, gotcha. Thx!